samples/bluetooth/hci_vs_scan_req/src/main.c

```diff
@@ -0,0 +1,353 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2024 Giancarlo Stasi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/bluetooth/addr.h>
+#include <host/hci_core.h>
+#include <common/bt_str.h>
+
+#if !defined(CONFIG_BT_EXT_ADV) && !defined(CONFIG_BT_CTLR_VS_SCAN_REQ_RX)
+#error "Either BT_EXT_ADV or BT_CTLR_VS_SCAN_REQ_RX must be selected to have scan request callback"
+#endif
+#if !defined(CONFIG_BT_HCI_VS_EVT_USER)
+#error "CONFIG_BT_HCI_VS_EVT_USER must be selected"
+#endif
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LENGTH (sizeof(DEVICE_NAME) - 1)
+
+/* Advertising Interval: the longer, the less energy consumption.
+ * Units: 0.625 milliseconds.
+ * The Minimum Advertising Interval and Maximum Advertising Interval should not be the same value
+ * (as stated in Bluetooth Core Spec 5.2, section 7.8.5)
+ */
+#define ADV_MIN_INTERVAL_625MS  BT_GAP_ADV_SLOW_INT_MIN
+#define ADV_MAX_INTERVAL_625MS  BT_GAP_ADV_SLOW_INT_MAX
+
+#if defined(CONFIG_BT_PERIPHERAL)
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_SCANNABLE | \
+				 BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#else
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_SCANNABLE | BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#endif
+
+/* Lower the value, closer the scanner. */
+#define ADV_TX_POWER RADIO_TXPOWER_TXPOWER_Neg8dBm
+
+static const struct bt_le_adv_param parameters = {
+	.options = ADV_OPTIONS,
+	.interval_min = ADV_MIN_INTERVAL_625MS,
+	.interval_max = ADV_MAX_INTERVAL_625MS,
+};
+
+static const struct bt_data advertisement_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LENGTH),
+};
+
+static const struct bt_data scan_response_data[] = {
+	BT_DATA_BYTES(BT_DATA_UUID128_ALL, 0),
+};
+
+#if defined(CONFIG_BT_EXT_ADV)
+static struct bt_le_ext_adv *ext_adv;
+#endif
```

Comment:
If the purpose of this sample is to showcase the VS specific functionality, I would just omit everything related to extended advertising to make it (way) simpler

Reply:
The different configurations have been left to experiment with them and compare performance and memory usage, mainly building with extended advertisements compared with VS scan and with broadcaster versus peripheral role.